### DISCUSSION TITLE
Are :dep crates cached?

### DISCUSSION THREAD

**Discussion by rmccampbell** [evcxr/evcxr#182] [Q&A] [answered]
When you run \:dep in a repl/notebook how are the dependent crates managed? Is it downloaded from scratch every session? Or is it cached somewhere? It seems to take a while to load crates even when they've been loaded in previous repls so I'm guessing the former. It would be nice to cache them in a documented location.

**Comment by bjorn3** [comment, marked as answer]
They are rebuild from scratch every session. The source files are cached by cargo though. What location would you suggest caching compiled dependencies?

**Reply by rmccampbell** [reply]
I guess the obvious answer is somewhere like ~/.evcxr. Maybe it could also use dependencies from a cargo workspace if run from within the workspace?

**Reply by bjorn3** [reply]
> I guess the obvious answer is somewhere like ~/.evcxr.

How would you handle cache invalidation then? Using a global location will likely cause the cache to grow forever consuming disk space without being clear to the user. It also wouldn't handle removal of notebooks.

**Reply by bjorn3** [reply]
The cache should be next to the notebook IMHO. It is up to @davidlattimore to make this decision though.

**Reply by rmccampbell** [reply]
For notebooks that would probably make sense, but maybe be configurable by env variable or something. For repls it would probably have to be opt-in because you don't want to create new cache folders every time you run the repl from a new folder.

**Reply by davidlattimore** [reply]
Evcxr supports using sccache to cache compilation outputs. See [setup instructions](https://github.com/google/evcxr/blob/main/COMMON.md#caching)

**Reply by rmccampbell** [reply]
Ah didn't see that, thanks.
